engine/crates/partial-caching/src/fetching/mod.rs

```diff
@@ -0,0 +1,98 @@
+//! This module handles the fetching phase of partial caching, where we're looking up things
+//! in the cache and refining the exeuction query further based on what we find.
+
+mod keys;
+
+use std::fmt;
+
+use common_types::auth::ExecutionAuth;
+use engine_value::Variables;
+use headers::HeaderMapExt;
+use runtime::cache::Entry;
+
+use self::keys::build_cache_keys;
+use super::CachingPlan;
+use crate::{execution::ExecutionPhase, CacheControlHeaders, FetchPhaseResult};
+
+impl CachingPlan {
+    pub fn start_fetch_phase(
+        self,
+        auth: &ExecutionAuth,
+        headers: &http::HeaderMap,
+        variables: &Variables,
+    ) -> CacheFetchPhase {
+        let cache_headers = headers
+            .typed_get::<headers::CacheControl>()
+            .unwrap_or_else(headers::CacheControl::new);
+
+        // TODO: don't forget to use CacheControlHeaders
+        CacheFetchPhase {
+            cache_keys: build_cache_keys(&self, auth, headers, variables),
+            cache_entries: std::iter::repeat_with(|| Entry::Miss)
+                .take(self.cache_partitions.len())
+                .collect(),
+            plan: self,
+            cache_headers,
+        }
+    }
+}
+
+/// This struct should be used to manage the cache fetching phase.
+pub struct CacheFetchPhase {
+    /// The CachingPlan that we're doing a fetch for.
+    pub(crate) plan: CachingPlan,
+
+    /// The keys for each cache_query in the plan.
+    ///
+    /// Will be None if we couldn't determine a key for whatever reason, in
+    /// which case we'll always query the executor for those fields
+    pub(crate) cache_keys: Vec<Option<String>>,
+
+    /// The cache control headers a user has provided
+    #[allow(unused)]
+    cache_headers: CacheControlHeaders,
```

Comment:
would have used `request_cache_control: RequestCacheControl` and `RequestCacheControl` could even be a new type to avoid mixups.

Reply:
Yeah, maybe.  I was kind of trying to get away from the phrase `cache_control` altogether.  It's a reasonable name, I was just finding the clash very confusing at points.

I think ideally I'd keep cache_control for this and have something different for the registry type, but urgh: that is a lot more work.

I'll give this some thought though, may change it.

Reply:
Was using `request_cache_control` for this in my refactor and `reponse_cache_control` for the response header IIRC. Not necessarily the best, but at least `request_cache_control` doesn't need a comment explaining what it is^^